apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -502,6 +504,7 @@ export function buildMachine(
             isPatDeviceConnected: false,
           }),
           target: 'pat_device_disconnected',
+          cond: (context) => context.isInVoterSession,
```

Comment:
I was thinking through whether we can avoid this small duplication of logic and I don't think we can in the current setup. I was imagining we'd check `auth` directly but I had forgotten that guards can't await promises: https://github.com/statelyai/xstate/issues/533

The post says the recommended workaround if you need to evaluate a promise as part of your guard is to use an intermediate state to contain the guard logic. This won't work in our case because the PAT events will continually fire and we'll ping pong between states constantly.

Sorry for not thinking of this rough edge before suggesting the change - given our options, this small duplication of logic is an ok tradeoff vs the larger task of restructuring when to invoke PAT polling.

Reply:
Yeah I did this workaround since I couldn't await in a cond.